### PR TITLE
ML/fix img cache

### PR DESCRIPTION
**Pull request by soupette**
#### Description of what you did:
This PR fixes the file caching and also prevent errors in the input file in case of errors

### CODE REVIEW

File: packages/strapi-helper-plugin/lib/src/utils/tests/prefixFileUrlWithBackendUrl.test.js [hunk 1]
```diff
@@ -14,4 +14,11 @@ describe('UPLOAD | utils | prefixFileUrlWithBackendUrl', () => {
 
     expect(prefixFileUrlWithBackendUrl(data)).toEqual(expected);
   });
+
+  it("should return the data if the url is not a string'", () => {
```
Comment (virginieky) [inline packages/strapi-helper-plugin/lib/src/utils/tests/prefixFileUrlWithBackendUrl.test.js]: additional apostrophe `string'`

File: packages/strapi-plugin-content-manager/admin/src/components/MediaPreviewList/index.js [hunk 2]
```diff
@@ -40,10 +40,18 @@ function MediaPreviewList({ hoverable, files }) {
 
     return (
       <MediaPreviewFile className={hoverable ? 'hoverable' : ''}>
-        <div>
-          <span>{fileType}</span>
-          <i className={`far fa-file-${fileType}`} />
-        </div>
+        {fileType ? (
+          <>
```
Comment (virginieky) [inline packages/strapi-plugin-content-manager/admin/src/components/MediaPreviewList/index.js]: The `Fragment` can be removed

File: packages/strapi-plugin-upload/admin/src/utils/tests/formatBytes.test.js [hunk 3]
```diff
@@ -7,27 +7,27 @@ describe('UPLOAD | components | EditForm | utils', () => {
     });
 
     it('should return 0B if less than 1 bytes is passed', () => {
-      expect(formatBytes(0.9)).toEqual('0B');
+      expect(formatBytes(0.9)).toEqual('900B');
     });
 
     it('should return 1KB if 1024 Bytes is passed', () => {
-      expect(formatBytes(1024)).toEqual('1KB');
+      expect(formatBytes(1024)).toEqual('1000KB');
```
Comment (HichamELBSI) [inline packages/strapi-plugin-upload/admin/src/utils/tests/formatBytes.test.js]: It seems that those tests are wrong. 1024B should equal 1KB. Why have they been updated?

File: packages/strapi-plugin-upload/admin/src/utils/tests/formatBytes.test.js [hunk 4]
```diff
@@ -7,27 +7,27 @@ describe('UPLOAD | components | EditForm | utils', () => {
...
     });
 
     it("should return 1KB if '1024' Bytes is passed", () => {
-      expect(formatBytes('1024')).toEqual('1KB');
+      expect(formatBytes('1024')).toEqual('1000KB');
```
Comment (HichamELBSI) [inline packages/strapi-plugin-upload/admin/src/utils/tests/formatBytes.test.js]: Is it a copy of the previous test?

File: packages/strapi-plugin-upload/admin/src/utils/tests/formatBytes.test.js [hunk 5]
```diff
@@ -7,27 +7,27 @@ describe('UPLOAD | components | EditForm | utils', () => {
     });
 
     it('should return 0B if less than 1 bytes is passed', () => {
-      expect(formatBytes(0.9)).toEqual('0B');
+      expect(formatBytes(0.9)).toEqual('900B');
```
Comment (HichamELBSI) [inline packages/strapi-plugin-upload/admin/src/utils/tests/formatBytes.test.js]: this is wrong. 0.9B is not equal to 900B

File: packages/strapi-plugin-upload/admin/src/utils/tests/formatBytes.test.js [hunk 6]
```diff
@@ -7,27 +7,27 @@ describe('UPLOAD | components | EditForm | utils', () => {
...
     });
 
     it("should return 1KB if '1024' Bytes is passed", () => {
-      expect(formatBytes('1024')).toEqual('1KB');
+      expect(formatBytes('1024')).toEqual('1000KB');
     });
 
     it('should return 1.21KB if 1034 Bytes is passed', () => {
-      expect(formatBytes(1234)).toEqual('1.21KB');
+      expect(formatBytes(1234)).toEqual('1.18MB');
```
Comment (HichamELBSI) [inline packages/strapi-plugin-upload/admin/src/utils/tests/formatBytes.test.js]: Can you update the test title. `it('should return 1.18KB ...`

File: packages/strapi-plugin-upload/admin/src/utils/tests/formatBytes.test.js [hunk 7]
```diff
@@ -7,27 +7,27 @@ describe('UPLOAD | components | EditForm | utils', () => {
...
     });
 
     it('should return 1.21KB if 1034 Bytes is passed with 3 decimals', () => {
-      expect(formatBytes(1234, 3)).toEqual('1.205KB');
+      expect(formatBytes(1234, 3)).toEqual('1.177MB');
     });
 
     it('should return 1 MB if 1.1e+6 Bytes is passed', () => {
-      expect(formatBytes(1100000, 0)).toEqual('1MB');
+      expect(formatBytes(1100000, 0)).toEqual('1GB');
```
Comment (HichamELBSI) [inline packages/strapi-plugin-upload/admin/src/utils/tests/formatBytes.test.js]: The test is wrong or the title should be updated.

File: packages/strapi-plugin-upload/admin/src/utils/tests/formatBytes.test.js [hunk 8]
```diff
@@ -7,27 +7,27 @@ describe('UPLOAD | components | EditForm | utils', () => {
...
     });
 
     it('should return 1.21KB if 1034 Bytes is passed', () => {
-      expect(formatBytes(1234)).toEqual('1.21KB');
+      expect(formatBytes(1234)).toEqual('1.18MB');
     });
 
     it('should return 1.21KB if 1034 Bytes is passed with 3 decimals', () => {
-      expect(formatBytes(1234, 3)).toEqual('1.205KB');
+      expect(formatBytes(1234, 3)).toEqual('1.177MB');
```
Comment (HichamELBSI) [inline packages/strapi-plugin-upload/admin/src/utils/tests/formatBytes.test.js]: Same here, we need some clarification about the `formatBytes` function.

File: packages/strapi-plugin-upload/admin/src/utils/tests/formatBytes.test.js (hunk 3 above)
Comment (soupette) [reply packages/strapi-plugin-upload/admin/src/utils/tests/formatBytes.test.js]: Because the backend sends the dimensions in kilo bits

**Pull request merged**